BeagleBone_black/src/sensors/vl6180.cpp

```diff
@@ -195,99 +221,39 @@ bool VL6180::isOnline()
 
 void VL6180::setContinuousRangingMode()
 {
-  if (continuous_mode_) {
-    log_.DBG("VL6180", "Sensor already in continuous ranging mode\n");
-    return;
-  }
   // Write to sensor and set to continuous ranging mode
   writeByte(kSysrangeStart, kModeStartStop | kModeContinuous);
   uint8_t inter_measurement_time = 1;
   writeByte(kSysrangeIntermeasurementPeriod, inter_measurement_time);
-  continuous_mode_ = true;
   log_.INFO("VL6180", "Sensor is in continuous ranging mode\n");
+  timeout_ = false;
 }
 
 uint8_t VL6180::continuousRangeDistance()
 {
-  uint8_t data;
-  data = 1;
+  uint64_t start = utils::Timer::getTimeMicros();
+  uint8_t data = 1;
+  uint8_t interrupt = 1;
+  uint64_t timeout = 50000;   // micro s
+  // Make sure we are in continuous ranging mode
+  readByte(kResultInterruptStatusGpio, &interrupt);
+
+  while (((interrupt & 0xC7) & 0x04) == 0) {
```

Comment:
`((interrupt & 0xC7) & 0x04)` is the same as `(interrupt & 0x04)`. Feel free to leave it as it is if there is any reason for it though. (e.g. 0xC7 somehow reflects the structure of the register)